src/main/scala/org/apache/spark/sql/pulsar/PulsarSource.scala

```diff
@@ -54,17 +68,32 @@ private[pulsar] class PulsarSource(
 
   private var currentTopicOffsets: Option[Map[String, MessageId]] = None
 
+
   private lazy val pulsarSchema: SchemaInfo = pulsarHelper.getPulsarSchema
 
   override def schema(): StructType = SchemaUtils.pulsarSourceSchema(pulsarSchema)
 
   override def getOffset: Option[Offset] = {
-    // Make sure initialTopicOffsets is initialized
+    throw new UnsupportedOperationException(
+      "latestOffset(Offset, ReadLimit) should be called instead of this method")
+  }
+
+  override def latestOffset(startingOffset: streaming.Offset,
+                            readLimit: ReadLimit): streaming.Offset = {
     initialTopicOffsets
-    val latest = pulsarHelper.fetchLatestOffsets()
-    currentTopicOffsets = Some(latest.topicOffsets)
-    logDebug(s"GetOffset: ${latest.topicOffsets.toSeq.map(_.toString).sorted}")
-    Some(latest.asInstanceOf[Offset])
+    val admissionLimits = AdmissionLimits(readLimit)
```

Comment:
This AdmissionLimits doesn't seem necessary to me.
Can we move the pattern match here to simplify things?
readLimit match {
    case ReadMaxBytes(maxBytes)  => pulsarHelper.latestOffsets(startingOffset, maxBytes)
    case ReadAllAvailable => pulsarHelper.fetchLatestOffsets()
  }